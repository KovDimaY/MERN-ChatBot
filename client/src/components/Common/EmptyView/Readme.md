EmptyView examples

```js
<EmptyView section="Experience" suggestion="Are you happy?" />
```
