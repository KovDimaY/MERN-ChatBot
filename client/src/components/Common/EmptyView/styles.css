.empty-view-container {
  max-width: 720px;
  margin: auto;
  text-align: center;
  font-size: 22px;
  padding: 25px 25px 100px 25px;
  color: #757575;
}
.empty-view-container p:first-child {
  font-size: 26px;
}

@media (max-width: 750px) {
  .empty-view-container {
    max-width: 450px;
    font-size: 18px;
    padding: 15px;
  }
  .empty-view-container p:first-child {
    font-size: 22px;
  }
}

@media (max-width: 500px) {
  .empty-view-container {
    max-width: 300px;
    font-size: 14px;
  }
  .empty-view-container p:first-child {
    font-size: 18px;
  }
}
