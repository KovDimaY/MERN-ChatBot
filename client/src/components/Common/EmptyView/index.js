import React from 'react';
import PropTypes from 'prop-types';

import './styles.css';

/**
 * This component was created to be shown when the page is not yet discovered at all (0%).
 * When at least something is discovered on the page it is going to be changed to
 * another view where some information is opened and some is hidden.
 */
const EmptyView = ({ section, children, suggestion }) => {
  const defaultDescription = (
    <React.Fragment>
      <p>This section is dedicated to the information about my {section}.</p>
      <p>
        For the moment you have not discovered anything regarding this topic. To
        get some info - ask the bot right question. When the bot gives you some
        related answer - the information will appear on this page. You will have
        a progress bar that will help you to understand how much info you have
        opened already.
      </p>
      <p>
        For example,{' '}
        <b>
          {'"'}
          {suggestion}
          {'"'}
        </b>{' '}
        sounds like a great question to start, but it can be something else too,
        just use your imagination ;)
      </p>
    </React.Fragment>
  );

  return (
    <div className="empty-view-container">{children || defaultDescription}</div>
  );
};

EmptyView.propTypes = {
  /** Name of the section to be added at the title of the page. */
  section: PropTypes.string.isRequired,
  /** Custom content of the page to overwrite the default one.  */
  children: PropTypes.node,
  /** This is an string with suggested question for the user.  */
  suggestion: PropTypes.string.isRequired,
};

EmptyView.defaultProps = {
  children: null,
};

export default EmptyView;
