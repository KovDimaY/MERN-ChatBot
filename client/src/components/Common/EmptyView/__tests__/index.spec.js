import React from 'react';
import { create } from 'react-test-renderer';

import EmptyView from '../../EmptyView';

describe('components/Common/<EmptyView />', () => {
  it('should render component without children', () => {
    const mockComponent = <EmptyView section="test" suggestion="test-suggestion" />;
    const tree = create(mockComponent).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with children', () => {
    const mockComponent = <EmptyView section="test">This is a test</EmptyView>;
    const tree = create(mockComponent).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
