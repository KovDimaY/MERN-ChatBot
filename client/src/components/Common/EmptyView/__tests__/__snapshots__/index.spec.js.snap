// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Common/<EmptyView /> should render component with children 1`] = `
<div
  className="empty-view-container"
>
  This is a test
</div>
`;

exports[`components/Common/<EmptyView /> should render component without children 1`] = `
<div
  className="empty-view-container"
>
  <p>
    This section is dedicated to the information about my 
    test
    .
  </p>
  <p>
    For the moment you have not discovered anything regarding this topic. To get some info - ask the bot right question. When the bot gives you some related answer - the information will appear on this page. You will have a progress bar that will help you to understand how much info you have opened already.
  </p>
  <p>
    For example,
     
    <b>
      "
      "
    </b>
     
    sounds like a great question to start, but it can be something else too, just use your imagination ;)
  </p>
</div>
`;
